oauth2/smart/patient-select.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/**
+ * patient-select.php
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <stephen@nielson.org>
+ * @copyright Copyright (c) 2021 Stephen Nielson <stephen@nielson.org>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Common\Session\SessionUtil;
+use OpenEMR\Core\Header;
+
+if ($oauthLogin !== true) {
+    $message = xlt("Error. Not authorized");
+    SessionUtil::oauthSessionCookieDestroy();
+    echo $message;
+    exit();
+}
+
+// make sure we have our patients set
+$errorMessage = $errorMessage ?? "";
+$patients = $patients ?? [];
+$redirect = $redirect ?? "";
+$searchAction = $searchAction ?? "";
+$fname = $searchParams['fname'] ?? "";
+$mname = $searchParams['mname'] ?? "";
+$lname = $searchParams['lname'] ?? "";
+$hasMore = $hasMore ?? false;
+
+?>
+<html>
+<head>
+    <title><?php echo xlt("OpenEMR Authorization"); ?></title>
+    <?php Header::setupHeader('opener'); ?>
+    <script>
+        (function(window) {
+            window.opener = null;
+        })(window || {});
+        <?php require($GLOBALS['srcdir'] . "/restoreSession.php"); ?>
+    </script>
+</head>
+<body class="container-fluid bg-dark">
+<div class="row h-100 w-100 justify-content-center align-items-center">
+    <div class="col-sm-6 bg-light text-dark">
+        <div class="text-md-center">
+            <h4 class="mb-4 mt-1"><?php echo xlt("Patient Selection"); ?></h4>
+        </div>
+        <div class="row w-100">
+            <div class="col">
+                <?php if (!empty($errorMessage)) : ?>
+                    <p class="alert alert-warning"><?php echo xlt($errorMessage); ?></p>
```

Comment:
where are these error messages coming from (just odd to translate them after the fact like this)?

Reply:
src/RestControllers/SMART/SMARTAuthorizationController.php line 213.